pkg/mcp/snapshot/snapshot.go

```diff
@@ -227,3 +239,54 @@ func (c *Cache) Status(group string) *StatusInfo {
 	}
 	return nil
 }
+
+// Types return all types of snapshots that the server layer is serving.
+func (c *Cache) Types() []string {
```

Comment:
You'll need to acquire the lock in both methods when accessing anything through c.*, Otherwise it will be a data race.